### PR TITLE
fix(warning): explicitly set key instead prop spreading

### PR DESCRIPTION
**Pull request by fbinz**
Spreading the key-prop leads to annoying warnings in development. This PR explicitly sets the property.

Relevant react PR: https://github.com/facebook/react/pull/25697

### CODE REVIEW

File: src/services/render/index.tsx [hunk 1]
```diff
@@ -80,24 +75,27 @@ export const renderElement = (
...
+        element,
+        onUpdate,
+        options: {
+          ...options,
+          inlineFormattingContext,
+        },
+        stylesheets,
+      };
+
       /* eslint-disable max-len */
```
Comment (flochtililoch) [inline src/services/render/index.tsx:89]: minor: can this comment be removed without causing new eslint errors? if not, could you re-enable it after the comments block (as previously done line 87)

File: src/services/render/index.tsx [hunk 2]
```diff
@@ -80,24 +75,27 @@ export const renderElement = (
...
-      // Object spreading will define the prop only when its value is truthy
-      /* eslint-enable max-len */
-      const extraProps: Partial<{ [key: string]: string | null }> = {};
-      if (element.getAttribute('key')) {
-        extraProps.key = element.getAttribute('key');
+      // Conditionally render the component with a key if it exists, to avoid
+      // warnings with current React versions, when the key attribute is set using the spread operator.
+      const key = element.getAttribute('key');
+
+      /* eslint-disable react/jsx-props-no-spreading */
```
Comment (flochtililoch) [inline src/services/render/index.tsx:94]: minor: either re-enable that rule after line 97, or explicitly disable both lines with `eslint-disable-line` comment.

**Review by flochtililoch — APPROVED** [review]
We appreciate your contribution @fbinz ! I added two minor comments related to eslint comments, would you have a couple minutes to address these before I merge? Thanks

**Comment by fbinz** [comment]
@flochtililoch 
Thanks for the review. I changed the two lines as suggested.

File: src/services/render/index.tsx [hunk 3]
```diff
@@ -80,24 +75,26 @@ export const renderElement = (
...
+        stylesheets,
+      };
+
+      // Conditionally render the component with a key if it exists, to avoid
+      // warnings with current React versions, when the key attribute is set
+      // using the spread operator.
+      const key = element.getAttribute('key');
+
+      if (key) {
+        <Component {...props} key={key} />; // eslint-disable-line react/jsx-props-no-spreading
```
Comment (flochtililoch) [inline src/services/render/index.tsx:95]: @fbinz I think you've accidentally removed the return statement here

Comment (fbinz) [reply src/services/render/index.tsx]: Oh, damn! Glad you caught that. Sorry for the troubles.

Comment (flochtililoch) [reply src/services/render/index.tsx]: thanks for the fix!

**Pull request merged**